62.CAD/main.cpp

```diff
@@ -15,20 +16,178 @@ enum class ExampleMode
 {
 	CASE_0, // Simple Line, Camera Zoom In/Out
 	CASE_1,	// Overdraw Fragment Shader Stress Test
-	CASE_2, // NOT USED
+	CASE_2, // hatches
 	CASE_3, // CURVES AND LINES
+	CASE_4, // STIPPLE PATTERN
 };
 
-constexpr ExampleMode mode = ExampleMode::CASE_3;
-using namespace nbl::hlsl;
+constexpr ExampleMode mode = ExampleMode::CASE_2;
+
+typedef uint32_t uint;
+
+// TODO: Use a math lib?
+double dot(const float64_t2& a, const float64_t2& b)
+{
+	return a.x * b.x + a.y * b.y;
+}
+
+double index(const float64_t2& vec, uint32_t index)
+{
+	const double* arr = &vec.x;
+	return arr[index];
+}
```

Comment:
remove, glm has both dot, and [] operator, and float64_t2 is glm vector on cpp now